src/operator/tensor/slice_sum-inl.h

```diff
@@ -0,0 +1,203 @@
+        idx = idx % stride;
+    }
+    in2_idx += idx + in2_offsets[ndim-1];
+    DType sum = in1_data[in1_idx] + in2_data[in2_idx];
+    KERNEL_ASSIGN(out_data[in1_idx], req, sum);
+  }
+};
+
+
+
+template<typename xpu>
+void SliceSumImpl(const nnvm::NodeAttrs& attrs,
+                      mshadow::Stream<xpu>* s,
+                      const TBlob& in1_data,
+                      const TBlob& in2_data,
+                      const OpReqType req,
+                      const TBlob& out_data) {
+  const SliceSumParam& param = nnvm::get<SliceSumParam>(attrs.parsed);
+  const int ndim_ = in1_data.ndim();
+  const int axis = param.axis;
+  int64_t begin = param.begin;
+  int64_t end = param.end;
+  if (end == -1) {
+    end = begin + in1_data.shape_[axis];
+  }
+
+  using namespace mxnet_op;
+  int64_t num_elements = out_data.Size();
+  CHECK_EQ (req, kWriteTo);
+
+  CHECK_LT (axis, ndim_);
+  CHECK_GE (begin, 0);
+  CHECK_GE (end, begin);
+  CHECK_LE (end, in1_data.shape_[axis]);
+  CHECK_EQ (end-begin, in1_data.shape_[axis]);
+
+  CHECK_EQ (in2_data.ndim(), ndim_);
+  for (int i = 0; i < ndim_; i++) {
+    if (i != axis) {
+      CHECK_EQ (in2_data.shape_[i], in1_data.shape_[i]);
+    } 
+  }
+
+  MXNET_NDIM_SWITCH(ndim_, ndim, {
+      common::StaticArray<int, ndim> in1_strides;
+      common::StaticArray<int, ndim> in2_offsets;
+      common::StaticArray<int, ndim> in2_strides;
+      in1_strides[ndim-1] = 1;
+      in2_strides[ndim-1] = 1;
+      if (ndim > 1) {
+        in1_strides[ndim-2] = in1_data.shape_[ndim-1];
+        in2_strides[ndim-2] = in2_data.shape_[ndim-1];
+      }
+
+      if (out_data.type_flag_ == mshadow::kFloat16) {
+        CHECK_EQ (num_elements % 2, 0);
+        CHECK_EQ (in1_data.shape_[ndim-1] % 2, 0);
+        CHECK_EQ (in2_data.shape_[ndim-1] % 2,  0);
+        num_elements /= 2;
+        if (axis == ndim - 1) {
+            if (begin >= 0) {
+                CHECK_EQ (begin % 2, 0);
+                begin /= 2;
+            }
+        }
+        if (ndim > 1) {
+          in1_strides[ndim-2] /= 2;
+          in2_strides[ndim-2] /= 2;
+        }
+      } 
+
+      for (int j = ndim-3; j >= 0; j--) {
+        in1_strides[j] = in1_strides[j+1] * in1_data.shape_[j+1];
+        in2_strides[j] = in2_strides[j+1] * in2_data.shape_[j+1];
+      }
+      for (int j = ndim-1; j >= 0; j--) {
+        if (j == axis) {
+            in2_offsets[j] = begin;
+        } else {
+            in2_offsets[j] = 0;
+        }
+      }
+
+      MSHADOW_TYPE_SWITCH_WITH_HALF2(out_data.type_flag_, DType, {
+        MXNET_ASSIGN_REQ_SWITCH(req, req_type, {
+          Kernel<SliceSumKernel<ndim, req_type>, xpu>::Launch(s, num_elements,
+              out_data.dptr<DType>(), in1_data.dptr<DType>(), in2_data.dptr<DType>(), 
+              in1_strides, in2_strides, in2_offsets);
+        })
+      })
+  })
+}
+
+template<typename xpu>
+void SliceSumOpForwardEx(const nnvm::NodeAttrs& attrs,
```

Comment:
nit: ending a name with Ex suggests that this function supports sparse inputs, which I don't think it does. Could you change it to just SliceSymOpForward?

Reply:
Fixed in new commit.